### ISSUE TITLE
Question On Potential Application

### ISSUE THREAD

**Issue by dfonsec** [VlSomers/keypoint_promptable_reidentification#15]
Hello! I've been trying to use the codebase for my own application. My data involves videos of professional baseball players performing baseball swings. I am using the Feature Extractor using the Solider (Partial Re-Id and OCC variants) model to try and identify the players. I haven't gained promising results as of yet. 

My goal is to have a system where you input a video (that my database of embeddings hasn't seen) of a professional player swing and it identifies that player based off the swing. Is this kind of thing possible using this codebase? What are things that I should consider?

**Comment by VlSomers** [comment]
Hi @dfonsec, it would be easier to help if you could show a few example images, or even some rankings/retrieval where it is possible to see how easy/hard your task is. Regarding pretrained weights, I would use the ones from PoseTrack21, since that dataset contains sports players similar to your use case. If the players your are trying to re-identify have exactly the same appearance, and you rather want to re-identify them based on their swing (i.e. movement), than you will probably need a ReID model that takes a video as input (KPR only takes images). You could even extract their poses (body keypoints) and re-identify them based on a short sequence of poses (that, again, would be a totally different model), with the assumption that every player has a unique style, i.e. a unique swing (for instance, I think it would be possible to identify tennis players solely using their skeletons). Let me know if you need more information.

**Comment by dfonsec** [comment]
Thank you for your reply @VlSomers. Here is an example of what I'm trying to achieve. Essentially, I want the model to be able to take in a [particular video of this player](https://www.youtube.com/shorts/flCYHk-rF6A) (Bichette) and then have it be able to identify that player based on training data of the same player ([so an example would be this video of the same player](https://www.youtube.com/shorts/IOVHHjDwHXc). Thank you for the advice and I'd love more information / pointers as to how you think I should proceed.

**Comment by VlSomers** [comment]
You should probably have a look at methods for video (aka tracklet) re-identification, that takes a short clip as input and output a single embedding, that can then be used to perform video retrieval in a database of videos. I will let you ask your favorite LLM about core concepts like representation learning, deep metric learning, image and video retrieval, embedding, cosine distance, etc. Then you should train that model on a large database of videos, with multiple athletes and multiple videos per athlete. You can for instance use the video ReID from Torchreid (https://github.com/KaiyangZhou/deep-person-reid), build a dataset with a similar structure to the MARS video reid dataset (https://kaiyangzhou.github.io/deep-person-reid/datasets.html#mars-mars), that you can easily plug into Torchreid, and then train your model.
